### PR TITLE
Add kubernetes.io managed groups for all remaining SIG/WGs

### PR DESCRIPTION
**Pull request by mrbobbytables**
This PR creates managed kubernetes.io groups for all remaining sigs/wgs that do not currently have them. Migrating to managed groups will resolve several issues we've had with  public googlegroups (e.g. mailing list limits on calendar invites or leads losing control of mailing lists), and enable us to use a common shared drive.

To complete this, I am adding myself to each new list to complete configuration (some settings can't currently be set via our automation). Once complete, I will remove myself from all the groups.

It also cleans up the general leads list and references each individual sig/wgs leads list so it won't have to be updated in more than one place. There are a few remaining emails in there that I'm waiting for acks on, but all in all its a significant improvement.

/hold

### CODE REVIEW

**Comment by mrbobbytables** [comment]
I forgot to update restrictions. Will do that now.

File: groups/sig-node/groups.yaml [hunk 1]
```diff
@@ -32,6 +32,29 @@ groups:
...
+      - decarr@redhat.com
+      - mpatel@redhat.com
+      - s.kanzhelev@gmail.com
+    settings:
+      WhoCanJoin: "ANYONE_CAN_JOIN"
+      WhoCanViewGroup: "ANYONE_CAN_VIEW"
+      WhoCanDiscoverGroup: "ANYONE_CAN_DISCOVER"
+      WhoCanPostMessage: "ANYONE_CAN_POST"
+      MessageModerationLevel: "MODERATE_NON_MEMBERS"
+      WhoCanViewMembership: "ALL_MANAGERS_CAN_VIEW"
```
Comment (SergeyKanzhelev) [inline groups/sig-node/groups.yaml:51]: Is there any issues with exposing the membership to members? In WG serving we found that without this setting members cannot invite the whole group to the meeting. Is there some privacy concerns to open membership view to everyone?

File: groups/sig-node/groups.yaml [hunk 2]
```diff
@@ -32,6 +32,29 @@ groups:
       - mpatel@redhat.com
       - s.kanzhelev@gmail.com
 
+  - email-id: sig-node@kubernetes.io
```
Comment (SergeyKanzhelev) [inline groups/sig-node/groups.yaml:35]: We will need one for CI group. Should I send a separate PR it you can help with it?

File: groups/sig-etcd/groups.yaml [hunk 3]
```diff
@@ -13,3 +13,21 @@ groups:
       - siarkowicz@google.com
       - spzala@us.ibm.com
       - wenjiazhang@google.com
+
+  - email-id: sig-etcd-leads@kubernetes.io
+    name: sig-etcd
```
Comment (soltysh) [inline groups/sig-etcd/groups.yaml:18]: Typo: should be sig-etcd-leads

File: groups/sig-architecture/groups.yaml [hunk 4]
```diff
@@ -19,6 +19,45 @@ groups:
   # subproject artifacts being stored in a given staging project
   #
 
+  - email-id: sig-architecture-leads@kubernetes.io
+    name: sig-architecture
```
Comment (soltysh) [inline groups/sig-architecture/groups.yaml:23]: Should be sig-architecture-leads

File: groups/sig-ui/groups.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+teams:
+  - email-id: sig-ui-leads@kubernetes.io
+    name: sig-ui
```
Comment (soltysh) [inline groups/sig-ui/groups.yaml:3]: sig-ui-leads

File: groups/wg-data-protection/groups.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+teams:
+  - email-id: wg-data-protection-leads@kubernetes.io
+    name: wg-data-protection
```
Comment (soltysh) [inline groups/wg-data-protection/groups.yaml:3]: wg-data-protection-leads

**Review by soltysh — APPROVED** [review]
A few minor comments, but big :+1:

**Comment by soltysh** [comment]
/hold 
so it doesn't merge by accident prematurely

File: groups/sig-node/groups.yaml (hunk 2 above)
Comment (mrbobbytables) [reply groups/sig-node/groups.yaml:35]: I'd do that separately as a follow up :+1: 
Hoping to get this merged relatively quickly

File: groups/sig-node/groups.yaml (hunk 1 above)
Comment (mrbobbytables) [reply groups/sig-node/groups.yaml:51]: Not really, it was a default from before. Good note on the invite thing, all the more reason to swap it.

**Review by soltysh — APPROVED** [review]
/lgtm

File: groups/sig-node/groups.yaml (hunk 1 above)
Comment (mrbobbytables) [reply groups/sig-node/groups.yaml:51]: I'm going to leave as is for now. I did verify that our other groups DO work with the current setting for sending invites, and when you open it up to everyone - all the details are present (person's email).

File: groups/sig-etcd/groups.yaml (hunk 3 above)
Comment (mrbobbytables) [reply groups/sig-etcd/groups.yaml]: fixed all of them :)

File: groups/sig-auth/groups.yaml [hunk 7]
```diff
@@ -19,6 +19,50 @@ groups:
...
+    description: |-
+      sig-auth leads
+    owners:
+      - killen.bob@gmail.com # temporary to finish configuration
+      - deads@redhat.com
+      - liggitt@google.com
+      - mikedanese@google.com
+      - rita.z.zhang@gmail.com
+      - theenjeru@gmail.com
+      - timallclair@gmail.com
```
Comment (enj) [inline groups/sig-auth/groups.yaml:33]: Tim is not a lead for Auth anymore

Comment (mrbobbytables) [reply groups/sig-auth/groups.yaml]: thanks, fixed. :+1:

**Comment by mrbobbytables** [comment]
We can also make changes to the various lists before completing migrations. Just want to get the scaffolding in there so we can move forward with next steps :+1:

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by ameukam** [comment]
@mrbobbytables this is a typo making the reconciliation fails:

```bash
2024/08/12 13:50:34 [unable to update group info for group "sig-instrumentation@kubernetes.io": googleapi: Error 503: Failed status in update settings response, backendError, unable to add bentheelder@google.com$ to "sig-testing-leads@kubernetes.io" as OWNER: googleapi: Error 400: Invalid Input: memberKey, invalid]
exit status 1
 ```

**Pull request merged**